src/Message/SmsMessageResultStatus.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Drupal\sms\Message;
+
+/**
+ * Defines states for SMS message results.
+ *
+ * Usually a setting a status on a result indicates something went wrong with
+ * the entire transaction.
+ */
+class SmsMessageResultStatus extends SmsMessageStatus{
+
+  /**
+   * Account error.
+   *
+   * Some configuration is required that can only be resolved on the gateway
+   * end.
+   */
+  const ACCOUNT_ERROR = 'account_error';
+
+  /**
+   * Too many requests.
+   */
+  const EXCESSIVE_REQUESTS = 'flooded';
+
+  /**
+   * Message could not be processed due to low credit.
+   */
+  const NO_CREDIT = 'no_credit';
+
+  /**
+   * Indicates the sender ID is invalid.
+   */
+  const INVALID_SENDER = 'invalid_sender';
+
+  /**
+   * Failed to authenticate with gateway.
+   */
+  const AUTHENTICATION = 'authentication';
+
+}
```

Comment:
Wouldnt a malformed URL problems be worked out when the gateway plugin is being implemented?

Do you have an example of language in the API docs of a SMS provider?

How about **missing parameters**?